### PR TITLE
Add option to return raw input stream for response

### PR DESCRIPTION
**Pull request by ajaaym**
Add option to return raw input stream for response

### CODE REVIEW

**Comment by ajaaym** [comment]
@chingor13 can you please review this, this is towards [3929](https://github.com/googleapis/google-cloud-java/issues/3929)

File: google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java [hunk 1]
```diff
@@ -239,6 +247,25 @@ public final boolean getDisableGZipContent() {
     return this;
   }
 
+  /**
+   * Sets whether the response should return raw input stream or not.
+   *
+   * <p>
+   * By default ti is {@code false}.
```
Comment (chingor13) [inline google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java]:
```suggestion
   * By default it is {@code false}.
```

File: google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java [hunk 2]
```diff
@@ -94,6 +94,9 @@
   /** Whether to disable GZip compression of HTTP content. */
   private boolean disableGZipContent;
 
+  /** whether to return raw input stream in {@link HttpResponse#getContent()}. */
```
Comment (chingor13) [inline google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java]:
```suggestion
  /** Whether to return raw input stream in {@link HttpResponse#getContent()}. */
```

File: google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java [hunk 3]
```diff
@@ -239,6 +247,25 @@ public final boolean getDisableGZipContent() {
...
 
+  /**
+   * Sets whether the response should return raw input stream or not.
+   *
+   * <p>
+   * By default ti is {@code false}.
+   * </p>
+   *
+   * <p>
+   * Normally response stream is wrapped GZipInputStream when it contains content-encoding header
```
Comment (chingor13) [inline google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java]: Maybe:

When the response contains a known content-encoding header, the response stream is wrapped with an InputStream that decodes the content. This fails when we download large files in chunks (see #bugnumber). Setting this to true will make the response return the raw input stream.

**Comment by ajaaym** [comment]
@chingor13 not sure whats going on with cla/google, can you check?

**Pull request merged**